### PR TITLE
Release v0.18.0

### PR DESCRIPTION
**Pull request by lizcross**
HOW TO TEST:
- copy paste the "finished tutorial code" that is in this PR into your `app.js` and make sure the tutorial runs as expected: https://github.com/mobify/astro/pull/753 http://localhost:9000/latest/guides/understanding-astro/first-simple-app/#final-solution

Update tutorial for Astro `0.18.0` release:
- iOS to Swift 3
- Android changes to support Android plugin refactor
- JavaScript build to use Webpack
- ES6 compatible JS, turned off linter rules that continuously break when using promises so that we don't need to change all of this to `sync`/`async` (including the corresponding docs)
- Updated astro docs to reflect changes and mostly to no longer break the linter when copy/pasting code snippets. See related PR: https://github.com/mobify/astro/pull/753

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## To be released
+
```
Comment (kerrmarin) [inline CHANGELOG.md]: Should this have an 0.18 somewhere?

File: app/app.js [hunk 2]
```diff
@@ -1,47 +1,37 @@
...
-        });
-
-    }, undefined, true);
+
+    // Enter your site url here
+    const baseUrl = 'http://<Your IP Address>:5000';
+
+    // Initialize plugins
+    const mainNavigationPromise = NavigationPlugin.init();
+    const layoutPromise = AnchoredLayoutPlugin.init();
```
Comment (kerrmarin) [inline app/app.js]: should we be using async/awaits?

File: app/config.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
 require.config({
```
Comment (kerrmarin) [inline app/config.js]: Is this file needed at all anymore? I thought with the new `import`s we didn't need this

File: ios/velo/AppDelegate.swift [hunk 4]
```diff
@@ -29,40 +29,40 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // run so that the AstroViewController can spin up and get app.js executing.
         // This allows any splash screen presented to be displayed before the Launch Image
         // is dismissed (at the end of didFinishLaunchingWithOptions).
-        NSRunLoop.mainRunLoop().runUntilDate(NSDate().dateByAddingTimeInterval(1.0))
+        RunLoop.main.run(until: Date().addingTimeInterval(1.0))
```
Comment (kerrmarin) [inline ios/velo/AppDelegate.swift]: I think we can get rid of this whole line now that we do the `dismissLaunchImage`

File: package.json [hunk 5]
```diff
@@ -3,18 +3,40 @@
   "version": "0.0.1",
   "author": "Mobify",
   "dependencies": {
-    "astro-sdk": "~0.17.0",
-    "bluebird": "~2.9.24"
+    "babel-runtime": "^6.20.0",
+    "bluebird": "~2.9.24",
+    "mobify-progressive-app-sdk": "~0.18.0",
+    "webpack": "^2.1.0-beta.27"
```
Comment (kerrmarin) [inline package.json]: Isn't webpack a dev dependency?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,5 @@
+## To be released
```
Comment (jeremywiebe) [inline CHANGELOG.md]: Are we missing changelog entries here (for changes in this PR)?

File: app/app.js (hunk 2 above)
Comment (jvoll) [inline app/app.js]: Going to hold off on this since the plan is for this tutorial to disappear... unless the team feels strongly otherwise of course.

File: package.json (hunk 5 above)
Comment (jvoll) [inline package.json]: YES. Oh man, I _think_ that this change fixed the back button issue I was experiencing last week (back button wouldn't show up in the header on iOS during navigation). OR, it's a race condition and I'm on the right side of the race right now. But, I think this fixed it so we'll go with that for now.
